app/src/main/java/com/example/movietrackerapp/view/MainActivity.kt

```diff
@@ -1,24 +1,55 @@
 package com.example.movietrackerapp.view
 
 import android.os.Bundle
+import android.widget.Toast
 import androidx.activity.enableEdgeToEdge
 import androidx.activity.viewModels
 import androidx.appcompat.app.AppCompatActivity
-import androidx.core.view.ViewCompat
-import androidx.core.view.WindowInsetsCompat
+import androidx.recyclerview.widget.RecyclerView
 import com.example.movietrackerapp.R
+import com.example.movietrackerapp.model.remote.service.response.MovieResponse
+import com.example.movietrackerapp.utils.ResponseWrapper
+import com.example.movietrackerapp.view.adapter.MovieAdapter
 import com.example.movietrackerapp.viewmodel.MovieServiceViewModel
 
 class MainActivity : AppCompatActivity() {
+    private lateinit var adapter: MovieAdapter
+    private lateinit var recyclerView: RecyclerView
     private val viewModel: MovieServiceViewModel by viewModels { MovieServiceViewModel.Factory }
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         enableEdgeToEdge()
         setContentView(R.layout.activity_main)
-        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
-            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
-            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
-            insets
+        setStateObservers()
+        viewModel.onGetRecommendations()
```

Comment:
for scalability you should create a method setEventListeners() and then call internally viewModel.onGetRecommendations